pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -120,3 +124,45 @@ func indexedDomainInterfaces(domain *api.Domain) map[string]api.Interface {
 	}
 	return domainInterfaces
 }
+
+// withNetworkIfacesResources adds network interfaces as placeholders to the domain spec
+// to trigger the addition of the dependent resources/devices (e.g. PCI controllers).
+// As its last step, it reads the generated configuration and removes the network interfaces
+// so none will be created with the domain creation.
+// The dependent devices are left in the configuration, to allow future hotplug.
+func withNetworkIfacesResources(vmi *v1.VirtualMachineInstance, domainSpec *api.DomainSpec, f func(v *v1.VirtualMachineInstance, s *api.DomainSpec) (cli.VirDomain, error)) (cli.VirDomain, error) {
+	domainSpecWithIfacesResource := domainSpec.DeepCopy()
+	requests := resource.ExtendedResourceList{ResourceList: vmi.Spec.Domain.Resources.Requests}
+	reqInterfaces := int(requests.Interface().Value())
+	for i := 0; i < (reqInterfaces - len(vmi.Spec.Domain.Devices.Interfaces)); i++ {
+		domainSpecWithIfacesResource.Devices.Interfaces = append(domainSpecWithIfacesResource.Devices.Interfaces, newInterfacePlaceholder(i))
+	}
```

Comment:
I think we can reduce the noise of domain spec with placeholders creation by extracting lines 134-139 to function.
IMO it can help reviewers focus on when to add placeholders logic.

Reply:
Done